src/main/java/org/symphonyoss/symphony/messageml/elements/PersonSelector.java

```diff
@@ -14,28 +21,109 @@
+
   private static final String PRESENTATIONML_TAG = "div";
-  private final static String MARKDOWN = "Person Selector";
+  private static final String PRESENTATIONML_PLACEHOLDER_ATTR = "data-placeholder";
+
+  private final static String MARKDOWN = "Person Selector:";
   private static final String CLASS_ATTR = "class";
+  private static final String PRESENTATIONML_NAME_ATTR = "data-name";
 
   public PersonSelector(Element parent) {
     super(parent, MESSAGEML_TAG);
   }
 
+  @Override
+  public void buildAll(MessageMLParser context, org.w3c.dom.Element element) throws InvalidInputException, ProcessingException {
+    switch (getFormat()) {
+      case MESSAGEML:
+        super.buildAll(context, element);
+        break;
+      case PRESENTATIONML:
+        buildElementFromPresentationML(context, element);
+        this.validate();
+        break;
+      default:
+        throw new InvalidInputException(String.format("Invalid message format for \"%s\" element", MESSAGEML_TAG));
+    }
+  }
+
   @Override
   public void validate() throws InvalidInputException {
     super.validate();
-    assertNoAttributes();
+
+    if (getAttribute(NAME_ATTR) == null) {
+      throw new InvalidInputException("The attribute \"name\" is required");
+    }
+    
     assertNoContent();
   }
 
   @Override
   public void asPresentationML(XmlPrintStream out) {
-    out.printElement(PRESENTATIONML_TAG, null, CLASS_ATTR, MESSAGEML_TAG);
+    Map<String, String> presentationAttrs = buildPersonSelectorInputAttributes();
+    out.printElement(PRESENTATIONML_TAG, presentationAttrs);
   }
 
   @Override
   public Node asMarkdown() {
-    return new FormElementNode(MARKDOWN);
+    return new FormElementNode(MARKDOWN, getAttribute(NAME_ATTR));
+  }
+
+  @Override
+  protected void buildAttribute(org.w3c.dom.Node item) throws InvalidInputException {
+    switch (item.getNodeName()) {
+      case NAME_ATTR:
+        setAttribute(NAME_ATTR, getStringAttribute(item));
+        break;
+      case PLACEHOLDER_ATTR:
+        setAttribute(PLACEHOLDER_ATTR, getStringAttribute(item));
+        break;
+      default:
+        throw new InvalidInputException("Attribute \"" + item.getNodeName()
+            + "\" is not allowed in \"" + getMessageMLTag() + "\"");
+    }
+  }
+
+  private Map<String, String> buildPersonSelectorInputAttributes() {
+    Map<String, String> presentationAttrs = new LinkedHashMap<>();
+    
+    presentationAttrs.put(CLASS_ATTR, MESSAGEML_TAG);
+    presentationAttrs.put(PRESENTATIONML_NAME_ATTR, getAttribute(NAME_ATTR));
+    
+    if(getAttribute(PLACEHOLDER_ATTR) != null) {
+      presentationAttrs.put(PRESENTATIONML_PLACEHOLDER_ATTR, getAttribute(PLACEHOLDER_ATTR));
+    }
+    
+    return presentationAttrs;
+  }
+
+  void buildElementFromPresentationML(MessageMLParser context, org.w3c.dom.Element element) throws InvalidInputException, ProcessingException {
+    
+    if (!"".equals(element.getAttribute(PRESENTATIONML_NAME_ATTR))) {
```

Comment:
Same here and in the check below about this check being needed. I think this whole if block can be a one liner without any issues.